pkg/kubelet/dockershim/docker_streaming.go

```diff
@@ -123,7 +125,12 @@ func (ds *dockerService) Exec(_ context.Context, req *runtimeapi.ExecRequest) (*
 	if err != nil {
 		return nil, err
 	}
-	return ds.streamingServer.GetExec(req)
+	r := (*runtimeapiv1.ExecRequest)(unsafe.Pointer(req))
```

Comment:
these are a copy/paste of the conversions in pkg/kubelet/cri/remote/conversion.go, right?

I don't love that these are disconnected from those unit tests, but I guess this is only until master reopens for 1.24

should we export those conversion methods from pkg/kubelet/cri/remote/conversion.go and use them here, or at least add comments to those conversion methods that if they have to change, these have to change as well?

Reply:
Since 1.23 will potentially be longer time supported version (at least by vendors) as the last one with dockershim, we will benefit from tests here. Maybe if this is too much change now we can bring them later?

Reply:
Yeah just copy pasted. I’d prefer not making the conversions Public because both (dockershim and v1alpha2) should be removed in the future.

I could follow-up on the tests since I’m running out of time for today, or add them Monday morning my time.

Reply:
I'd be ok with either exporting the conversion methods that are already tested and using them here, or commenting in the conversion methods that this code needs to be adjusted if the conversion is adjusted

Reply:
the comment seems fine, given the lifetime of this code is ~1 week in master

Reply:
Adding the comment in the next minutes.

Reply:
Added the comments on both sides of the implementation.